spec/lib/submission_builder/ty2024/states/md/documents/md502_spec.rb

```diff
@@ -657,5 +659,16 @@
         expect(xml.at("Form502 RefundableTaxCredits")&.text).to eq('400')
       end
     end
+
+    context "when there are taxes owed" do
+      before do
+        allow(intake).to receive(:refund_or_owe_taxes_type).and_return(:owe)
+        allow(intake).to receive(:payment_or_deposit_type).and_return(:direct_deposit)
+      end
+
+      it 'does not include the NameOnBankAccount attribute' do
+        expect(xml.at("Form502 NameOnBankAccount")).not_to be_present
+      end
+    end
```

Comment:
can we move this spec to the
```
    context "Line 51d: NameOnBankAccount" do
```
and test that joint account holder name also does not show up as well?

Reply:
yes have done!